lms/djangoapps/courseware/views/views.py

```diff
@@ -733,6 +734,13 @@ def create_page_context(self, request, course=None, tab=None, **kwargs):
             'openedx.course_experience.reset_course_deadlines', kwargs={'course_id': text_type(course.id)}
         )
 
+        reset_deadlines_banner_can_display = False
+        course_overview = CourseOverview.objects.get(id=course.id)
+        if timezone.now() < course_overview.end_date:
+            course_enrollment = CourseEnrollment.objects.filter(course=course_overview, user=request.user).first()
+            if course_enrollment and course_enrollment.mode == CourseMode.VERIFIED:
```

Comment:
Should this also have a check on the Relative Dates flag?

Reply:
ohhhhh i like that